plugins/modules/zos_copy.py

```diff
@@ -1581,15 +1592,55 @@ def restore_backup(dest, backup, dest_type, use_backup, volume=None):
                 shutil.rmtree(dest, ignore_errors=True)
                 shutil.copytree(backup, dest)
         else:
-            data_set.DataSet.ensure_absent(dest, volumes)
-
             if dest_type in data_set.DataSet.MVS_VSAM:
+                data_set.DataSet.ensure_absent(dest, volumes)
                 repro_cmd = """  REPRO -
                 INDATASET('{0}') -
                 OUTDATASET('{1}')""".format(backup.upper(), dest.upper())
                 idcams(repro_cmd, authorized=True)
+            elif dest_type in data_set.DataSet.MVS_SEQ:
+                response = datasets._copy(backup, dest)
+                if response.rc != 0:
+                    raise CopyOperationError(
+                        "An error ocurred while restoring {0} from {1}".format(dest, backup),
+                        response.rc,
+                        response.stdout_response,
+                        response.stderr_response
+                    )
             else:
-                datasets.copy(backup, dest)
+                if not members_to_restore:
+                    members_to_restore = []
+                if not members_to_delete:
+                    members_to_delete = []
+
+                for member in members_to_restore:
+                    response = datasets._copy(
+                        "{0}({1})".format(backup, member),
+                        "{0}({1})".format(dest, member)
+                    )
+
+                    if response.rc != 0:
+                        raise CopyOperationError(
```

Comment:
Maybe it would be a nice addition to, in case of error restoring members return a list of members restored and not restored. @rexemin Let me know your comments.

Reply:
I agree, that would help users understand better in what state their system is after the module fails. I'll add the lists to the error message.